src/tactic/solve_by_elim.lean

```diff
@@ -193,7 +218,8 @@ do
   (if opt.backtrack_all_goals then id else focus1) $ (do
     gs ← get_goals,
     solve_by_elim_aux opt.to_basic_opt gs lemmas opt.max_depth <|>
-    fail "solve_by_elim failed; try increasing `max_depth`?")
+    fail ("`solve_by_elim` failed.\nTry `solve_by_elim { max_depth := N }` for a larger `N`,\n" ++
```

Comment:
Maybe mention the default (or current) value of `N` here?

Reply:
Good idea. It now says "for `N > 3`" (or whatever the actual value is).